modulo-1-fundamentos/bloco-8-higher-order-functions-do-javascript-es6/dia-1-javascript-es6-introducao-a-higher-order-functions/bonus.js

```diff
@@ -0,0 +1,107 @@
+const mage = {
+  healthPoints: 130 + Math.floor(Math.random() * 100),
+  intelligence: 45,
+  mana: 15 + Math.floor(Math.random() * 50),
+  damage: undefined,
+};
+
+const warrior = {
+  healthPoints: 200,
+  strength: 30,
+  weaponDmg: 2,
+  damage: undefined,
+};
+
+const dragon = {
+  healthPoints: 300 + Math.floor(Math.random() * 100),
+  strength: 50 + Math.floor(Math.random() * 50),
+  damage: undefined,
+};
+
+const battleMembers = { mage, warrior, dragon };
+
+const handleDragonDamage = () =>
+  (dragon.damage = Math.floor(Math.random() * (dragon.strength - 15 + 1) + 15));
+
+const handleWarriorDamage = () =>
+  (warrior.damage = Math.floor(
```

Comment:
Ainda vamos ver na Trybe, mas já da uma olhada em desestruturação de objetos, pois ajuda bastante e deixa seu código mais limpo.
```suggestion
   const { strength, weaponDmg, damage } = warrior;
   (damage = Math.floor(
   Math.random() *
      (weaponDmg * strength - strength + 1) +
      strength
  ));
```

Reply:
Tava lendo a respeito ontem, útil demais.